include/secp256k1_recovery.h

```diff
@@ -92,7 +92,22 @@ SECP256K1_API int secp256k1_ecdsa_sign_recoverable(
 
 /** Recover an ECDSA public key from a signature.
  *
- *  Returns: 1: public key successfully recovered (which guarantees a correct signature).
+ *  Note that successful public key recovery implies a correct ECDSA signature
+ *  in a mathematical sense, but it is only guaranteed to also pass verification
+ *  with `secp256k1_ecdsa_verify` if it is in lower-S form. For verifying
```

Comment:
"correct ECDSA signature in a mathematical sense" sounds a bit imprecise. How about the following wording, which avoids that term entirely?

```
Successful public key recovery guarantees that the signature, in normalized
form, passes `secp256k1_ecdsa_verify`. Thus, explicit verification is not
necessary.

A recoverable signature converted to a normal signature may not pass
`secp256k1_ecdsa_verify` if it is not normalized. If a normalized signature is
required, call `secp256k1_ecdsa_signature_normalize` after
`secp256k1_ecdsa_recover`.
```

Reply:
Took that suggestion, and set you as commit author accordingly. As tiny-nit, it's slightly unfortunate that "normal" and "normalized" sound so similar, but not sure how to avoid it (maybe it's not a big deal).

Reply:
SGTM, taken.

Reply:
"a non-recoverable signature [...] (even if it passes `secp256k1_ecdsa_recover`)" may be a bit confusing because a non-recoverable signature cannot pass `secp256k1_ecdsa_recover`.

How about this:

```
However, a recoverable signature that successfully passes `secp256k1_ecdsa_recover`,
when converted to a non-recoverable signature (using
`secp256k1_ecdsa_recoverable_signature_convert`), is not guaranteed to be
normalized and thus not guaranteed to pass `secp256k1_ecdsa_verify`.
```

Reply:
Updated again with the suggestion.